json-format.md

```diff
@@ -211,6 +260,63 @@ or [JSON data](#31-handling-of-data) data:
 }
 ```
 
+The above example encoded using [HTTP Binary Content Mode][http-binary]:
+
+```
+CE-Specversion: 1.0
+CE-Type: com.example.someevent
+CE-Source: /mycontext
+CE-Id: C234-1234-1234
+CE-Time: 2018-04-05T17:31:00Z
+CE-Comexampleextension1: value
+CE-Comexampleothervalue: 5
+Content-Type: application/json
+
+{
+  "appinfoA" : "abc",
+  "appinfoB" : 123,
+  "appinfoC" : true
+}
+```
+
+Example event with non-`Binary`-valued `data` and an implicit `datacontenttype`
+of `application/json`. The data is a literal JSON string:
+
+```JSON
+{
+    "specversion" : "1.0",
+    "type" : "com.example.someevent",
+    "source" : "/mycontext",
+    "subject": null,
+    "id" : "D234-1234-1234",
+    "time" : "2018-04-05T17:31:00Z",
+    "comexampleextension1" : "value",
+    "comexampleothervalue" : 5,
+    "data" : "I'm just a string"
+}
+```
+
+The above example encoded using [HTTP Binary Content Mode][http-binary]:
+
+```
+CE-Specversion: 1.0
+CE-Type: com.example.someevent
+CE-Source: /mycontext
+CE-Id: D234-1234-1234
+CE-Time: 2018-04-05T17:31:00Z
+CE-Comexampleextension1: value
+CE-Comexampleothervalue: 5
+Content-Type: application/json
```

Comment:
Yes, this is a deliberate difference, as noted in the prose below. In JSON structured format, the data content type is implicitly `application/json` if the `datacontenttype` field is absent, as [stated in the core spec](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#datacontenttype). However, this is _not_ true for HTTP binary. Thus, in order to represent the _same event_ as the earlier JSON structured event, we MUST introduce a Content-Type header in Binary mode.

Reply:
I'm not sure what you mean by the "event format" being relevant for binary mode, or that the spec implies that it shouldn't (be relevant?)

Reply:
I see. Yes, I agree, the fact that the event previously was encoded in a certain event format (in this case JSON structured mode) should be irrelevant to its later re-encoding in binary mode. The binary mode encoding of an event should _not_ include information that the event "originally came from" a certain event format. However, that's not what's going on here. Here, the type of the _event payload_ (i.e. the event's "data" attribute) is a JSON value. It's not simply a string (which we could have, for example, re-encoded as text/plain), but it is specifically a _JSON String_, which is additional information that could effect the in-memory representation or runtime semantics of the event. Importantly, this is independent of the fact that the event happened to have been encoded previously in JSON structured mode. We are adding a Content-Type here _not_ to encode the fact that the event was previously JSON-formatted, but to preserve the event semantics by preserving the _type of the payload_.

Reply:
Right. This is why I think we shouldn't be defaulting at all, ideally - we're in limbo. Is the data content type of the CloudEvent "application/json" or not? If it is, it should be treated that way consistently. If it's not, we shouldn't be adding it in. <sigh>

Ideally I'd really like datacontenttype to be required whenever data is populated, but it's *much* too late to impose that now...

Reply:
Yeah, `datacontenttype` probably should have been required if data is present, but that's long under the bridge now. For now, to conform with what [spec.md says](https://github.com/cloudevents/spec/blob/master/spec.md#datacontenttype), we have to allow for implicit behavior on _deserialization_ of JSON format, AND we have to recommend that the implicit value be set explicitly when moving to a different format. So that's what we have here.